version: '3.7'

services:
  db:
    image: postgres
    healthcheck:
      test: pg_isready --dbname=PGdb --host=localhost --port=5432 --username=mazino
      interval: 40s
      timeout: 5s
      retries: 10
    restart: always
    ports:
      - "5432:5432"
    environment:
      POSTGRES_PASSWORD: mazino
      POSTGRES_USER: mazino
      POSTGRES_DB: PGdb
    networks:
      - my-network

  back:
    build:
      context: ./back
      dockerfile: Dockerfile
    healthcheck:
      test: curl --fail http://localhost:4000/ || exit 1
      interval: 40s
      timeout: 5s
      retries: 10
    ports:
      - "4000:4000"
    env_file:
      - back/.env
    depends_on:
      db:
        condition: service_healthy
    networks:
      - my-network

  front:
    build:
      context: ./front
      dockerfile: Dockerfile
    ports:
      - "3000:3000"
    environment:
      - BACK_IP=back
    depends_on:
      back:
        condition: service_healthy
    networks:
      - my-network

networks:
  my-network:
    driver: bridge
